ChromaticHQ/move-to-next-iteration
# Add support for optional excluded-statuses option

## Description

This PR updates our fork of the `move-to-next-iteration` action to validate it works as intended before submitting a PR upstream. This action can be used to automatically move tickets from one the previous iteration to the current iteration, which is helpful on projects where tickets often roll over from one sprint to the next.

The purpose of the changes in this PR is to add support for selecting tickets via an `excluded-statuses` setting, as opposed to a `statuses` setting. This makes it easier to configure and maintain workflows that use this action on projects where the desire is to move all tickets to a new sprint except for the ones that match one or more statuses.

## Motivation / Context

blombard/move-to-next-iteration#5

## Testing Instructions / How This Has Been Tested

I tested this on one of our project repos by adding and configuring a workflow that can be run manually. I then moved a backlog ticket into the last sprint, ran the workflow, and confirmed that the ticket now appears in the current sprint with no other side-effects. See screenshots below.

## Screenshots

Before Workflow Run|After Workflow Run
---|---
![pre-workflow](https://github.com/ChromaticHQ/move-to-next-iteration/assets/439649/2b7e9b55-187e-4224-a93b-e871aed36d74)|![post-workflow](https://github.com/ChromaticHQ/move-to-next-iteration/assets/439649/9413ade0-619e-4c85-aba9-f745c0181412)
Notice ticket 2839 is in Sprint 31 and Sprint 32 shows 9 open tickets.|Ticket 2839 is in Sprint 32, which now shows 10 open tickets. Sprint 31 no longer shows up in this view, indicating there are no tickets in it.

## Documentation

I updated the README and added code comments where it seemed appropriate.
